Modules/mathmodule.c

```diff
@@ -2309,7 +2309,7 @@ loghelper(PyObject* arg, double (*func)(double))
             assert(e >= 0);
             assert(!PyErr_Occurred());
             /* Value is ~= x * 2**e, so the log ~= log(x) + log(2) * e. */
-            result = func(x) + func(2.0) * e;
+            result = fma(func(2.0), (double)e, func(x));
```

Comment:
Is this type cast required?  I don't think that compiler will warn about lossy conversion here.  (Just as before in `func(2.0)*e`.)

Reply:
Ah, my fault.  I forgot that `-Wconversion` is not enabled by `-Wpedantic`.  Though, if it is - this will issue warnings in *both* cases, i.e. for arithmetic too.  Strange Microsoft products...